src/igr-tests/igr-runner.cc

```diff
@@ -47,52 +45,91 @@ void cycles_test();
 
 int main(int argc, char **argv)
 {
-    void (*test_funcs[])() = { basic_test, environ_test, environ2_test, ps_environ_test,
-            chain_to_test, force_stop_test, restart_test, check_basic_test, check_cycle_test,
-            check_cycle2_test, check_lint_test, reload1_test, reload2_test, no_command_error_test,
-            add_rm_dep_test, var_subst_test, svc_start_fail_test, dep_not_found_test,
-            pseudo_cycle_test, before_after_test, before_after2_test, log_via_pipe_test,
-            catlog_test, offline_enable_test, xdg_config_test, cycles_test };
+    struct test tests[] = { { "basic", basic_test }, { "environ", environ_test },
+            { "environ2", environ2_test }, { "ps-environ", ps_environ_test },
+            { "chain-to", chain_to_test }, { "force-stop", force_stop_test },
+            { "restart", restart_test }, { "check-basic", check_basic_test },
+            { "check-cycle", check_cycle_test }, { "check-cycle2", check_cycle2_test },
+            { "check-lint", check_lint_test }, { "reload1", reload1_test },
+            { "reload2", reload2_test }, { "no-command-error", no_command_error_test },
+            { "add-rm-dep", add_rm_dep_test }, { "var-subst", var_subst_test },
+            { "svc-start-fail", svc_start_fail_test, }, { "dep-not-found", dep_not_found_test },
+            { "pseudo-cycle", pseudo_cycle_test }, { "before-after", before_after_test},
+            { "before-after2", before_after2_test }, { "log-via-pipe", log_via_pipe_test },
+            { "catlog", catlog_test }, { "offline-enable", offline_enable_test },
+            { "xdg-config", xdg_config_test }, { "cycles", cycles_test } };
     const char * const test_dirs[] = { "basic", "environ", "environ2", "ps-environ", "chain-to", "force-stop",
+            "restart", "check-basic", "check-cycle", "check-cycle2", "check-lint", "reload1", "reload2",
```

Comment:
Why are these lines duplicated?

Reply:
So it was just a mistake? Please communicate :)

Reply:
Sorry, Sometimes I think I was "blind" when reviewed my code :((